server/api/src/rsDataObjTrim.cpp

```diff
@@ -116,38 +152,42 @@ namespace
         }
     } // get_data_object_info
 
-    auto get_list_of_replicas_to_trim(
-        dataObjInp_t& _inp,
-        const irods::file_object_ptr _obj,
-        const DataObjInfo& _info)
+    auto get_list_of_replicas_to_trim(const DataObjInp& _inp,
+                                      const irods::file_object_ptr& _obj,
+                                      const DataObjInfo& _info) -> std::vector<irods::physical_object>
     {
-        const auto& list = _obj->replicas();
-
-        std::vector<irods::physical_object> trim_list;
-
-        const unsigned long good_replica_count = std::count_if(list.begin(), list.end(),
-            [](const auto& repl) {
-                return GOOD_REPLICA == repl.replica_status();
-            });
-
         const auto minimum_replica_count = get_minimum_replica_count(getValByKey(&_inp.condInput, COPIES_KW));
-        const auto expiration = get_time_of_expiration(getValByKey(&_inp.condInput, AGE_KW));
-        const auto expired = [&expiration](const irods::physical_object& obj) {
-            return expiration && std::atoi(obj.modify_ts().c_str()) > expiration;
+        const auto& replica_list = _obj->replicas();
+        auto remaining_replica_count = replica_list.size();
+        if (std::cmp_less_equal(remaining_replica_count, minimum_replica_count)) {
+            // If the number of existing replicas is already at the minimum required count, we should return an empty
+            // list so that no replicas are trimmed.
+            return {};
+        }
+
+        const auto good_replica_count = std::count_if(replica_list.begin(), replica_list.end(), [](const auto& repl) {
+            return GOOD_REPLICA == repl.replica_status();
+        });
+        constexpr decltype(good_replica_count) minimum_good_replica_count = 1;
+
+        const auto* age_kw = getValByKey(&_inp.condInput, AGE_KW);
+        const auto expiration_time = get_time_of_expiration(age_kw);
+        const auto expired = [&expiration_time](const irods::physical_object& _replica) {
```

Comment:
Copying the expiration time into the lambda is likely better since it's a readonly value.

The compiler may be able to optimize the code more since it wouldn't need to deal with the indirection introduced by the reference to `expiration_time`.

Reply:
Will update